exchanges/exchange.go

```diff
@@ -309,68 +309,56 @@ func (b *Base) SetCurrencyPairFormat() error {
 func (b *Base) SetConfigPairs() error {
 	assetTypes := b.Config.CurrencyPairs.GetAssetTypes(false)
 	exchangeAssets := b.CurrencyPairs.GetAssetTypes(false)
-	for x := range assetTypes {
-		if !exchangeAssets.Contains(assetTypes[x]) {
-			log.Warnf(log.ExchangeSys,
-				"%s exchange asset type %s unsupported, please manually remove from configuration",
-				b.Name,
-				assetTypes[x])
-			continue // If there are unsupported assets contained in config, skip.
-		}
-
-		var enabledAsset bool
-		if b.Config.CurrencyPairs.IsAssetEnabled(assetTypes[x]) == nil {
-			enabledAsset = true
+	for _, a := range assetTypes {
+		if !exchangeAssets.Contains(a) {
+			log.Warnf(log.ExchangeSys, "%s exchange asset type %s unsupported, please manually remove from configuration", b.Name, a)
+			continue
 		}
-
-		err := b.CurrencyPairs.SetAssetEnabled(assetTypes[x], enabledAsset)
-		// Suppress error when assets are enabled by default and they are being
-		// enabled by config. A check for the inverse
-		// e.g. currency.ErrAssetAlreadyDisabled is not needed.
-		if err != nil && !errors.Is(err, currency.ErrAssetAlreadyEnabled) {
-			return err
+		if !b.Config.CurrencyPairs.UseGlobalFormat {
+			// TODO: Should probably be in SetCurrencyPairFormat. See #1748
```

Comment:
_should_? _probably_? No confidence in this direction? Remove that double-weak wording 💪

Reply:
🫡 

Fixed https://github.com/gbjk/gocryptotrader/commit/d01aa79f92